javascript/javascript2/week2/homework/hyf-bay/main.js

```diff
@@ -0,0 +1,94 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+const productsList = document.getElementById("productList");
+console.log(products);
+
+//creates and adds INPUT elements to html document
+const searchPanel = document.getElementById("searchPanel");
+const searchProduct = document.createElement("input");
+searchProduct.setAttribute("type", "text");
+searchProduct.setAttribute("id", "searchProduct");
+searchProduct.setAttribute("placeholder", "Product name");
+searchPanel.appendChild(searchProduct);
+
+const searchPrice = document.createElement("input");
+searchPrice.setAttribute("type", "text");
+searchPrice.setAttribute("id", "searchPrice");
+searchPrice.setAttribute("placeholder", "Product price");
+searchPanel.appendChild(searchPrice);
+
+function renderProducts(products) {
+  products.forEach((product) => {
+    const productInfo = document.createElement("li");
+    productInfo.innerHTML = `
+            <ul class="product-info">
+                <li class="product-name">${product.name}</li>
+                <li>Price: ${product.price}</li>
+                <li>Rating: ${product.rating}</li>
+            </ul>
+        `;
+    productsList.appendChild(productInfo);
+  });
+}
+renderProducts(products);
+
+//searching by product name
+searchProduct.addEventListener(`input`, function (event) {
+  const inputProductName = event.target.value;
+  const filteredProductByName = products.filter((product) =>
+    product.name.toLowerCase().indexOf(inputProductName.toLowerCase() != -1)
+  );
+  renderProducts(filteredProductByName);
+});
+
+//searching by product price
+searchPrice.addEventListener("input", function (event) {
+  const inputProductPrice = event.target.value;
+  const filteredProductByPrice = products.filter(function (product) {
+    return product.price <= parseInt(inputProductPrice);
+  });
```

Comment:
Again you should delete the previous list and you don't need to use "parseInt".

//searching by product price
  searchPrice.addEventListener("input", function (event) {
  const inputProductPrice = event.target.value;
  const filteredProductByPrice = products.filter(function (product) {
       return product.price <= inputProductPrice;
  });
  productsList.innerHTML = " ";
  renderProducts(filteredProductByPrice);
});